// react-message-context/dist/message-provider.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _messageContext = require('./message-context');

var _propTypes3 = require('./prop-types');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var MessageProvider = function (_Component) {
  _inherits(MessageProvider, _Component);

  function MessageProvider() {
    var _ref;

    var _temp, _this, _ret;

    _classCallCheck(this, MessageProvider);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return _ret = (_temp = (_this = _possibleConstructorReturn(this, (_ref = MessageProvider.__proto__ || Object.getPrototypeOf(MessageProvider)).call.apply(_ref, [this].concat(args))), _this), _this.state = _messageContext.defaultValue, _temp), _possibleConstructorReturn(_this, _ret);
  }

  _createClass(MessageProvider, [{
    key: 'render',
    value: function render() {
      return _react2.default.createElement(
        _messageContext.Provider,
        { value: this.state },
        this.props.children
      );
    }
  }], [{
    key: 'getDerivedStateFromProps',
    value: function getDerivedStateFromProps(_ref2, prev) {
      var fallback = _ref2.fallback,
          locale = _ref2.locale,
          messages = _ref2.messages,
          pathSep = _ref2.pathSep;

      if (typeof pathSep !== 'string') pathSep = pathSep ? MessageProvider.defaultProps.pathSep : null;
      var locales = fallback ? [locale].concat(fallback) : locale ? [locale] : [];
      return messages !== prev.messages || pathSep !== prev.pathSep || locales.length !== prev.locales.length || prev.locales.some(function (lc, i) {
        return lc !== locales[i];
      }) ? { locales: locales, messages: messages, pathSep: pathSep } : null;
    }
  }]);

  return MessageProvider;
}(_react.Component);

MessageProvider.propTypes = {
  fallback: _propTypes3.PathType,
  locale: _propTypes2.default.string,
  messages: _propTypes2.default.object.isRequired,
  pathSep: _propTypes2.default.oneOfType([_propTypes2.default.bool, _propTypes2.default.string])
};
MessageProvider.defaultProps = {
  pathSep: '.'
};
exports.default = MessageProvider;
